service/src/main/java/bio/terra/landingzone/stairway/flight/create/CreateLandingZoneFlight.java

```diff
@@ -40,16 +41,37 @@ private void addCreateSteps(LandingZoneFlightBeanBag flightBeanBag, FlightMap in
       throw new LandingZoneCreateException("Unable to find requested landing zone in input map");
     }
 
+    var landingZoneId = inputParameters.get(LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+    if (landingZoneId == null) {
+      throw new LandingZoneCreateException("Landing zone identifier is not defined.");
+    }
+
     addStep(
         new CreateSamResourceStep(flightBeanBag.getSamService()), RetryRules.shortExponential());
 
     addStep(
         new GetBillingProfileStep(flightBeanBag.getBpmService()), RetryRules.shortExponential());
 
     if (!requestedLandingZone.isAttaching()) {
-      addStep(
-          new CreateAzureLandingZoneStep(flightBeanBag.getAzureLandingZoneManagerProvider()),
-          RetryRules.cloud());
+      // TODO: use step which run sub-flight with all these steps
+      // so, in this case we need only one step to rollback everything at once
+      // and don't need to implement rollback for each step
+      // Which option is better here 1 rollback or rollback per resource creation?
```

Comment:
IMO a single step rollback is simpler (if perhaps a bit less obvious). If we really need to we can switch to rollback-per-step.

Reply:
Current approach guarantees the right order of undo. And yes, it's more obvious - each step incorporates compensating transaction.